.github/workflows/compilation_on_nuttx.yml

```diff
@@ -43,20 +43,36 @@ jobs:
     strategy:
       matrix:
         nuttx_board_config : [
+          # x64
+          "sim:nsh",
+          # cortex-m0
+          "raspberrypi-pico:nsh",
+          # cortex-m4
           "stm32f4discovery:nsh",
+          # cortex-m7
           "nucleo-h743zi:nsh",
+          # cortex-a9
           "sabre-6quad:nsh",
+          # riscv32imac
+          "rv-virt:nsh",
+          # riscv64imac
+          "rv-virt:nsh64",
+          # riscv64gc
+          "maix-bit:nsh",
         ]
         wamr_config_option: [
           "CONFIG_INTERPRETERS_WAMR=y\\nCONFIG_INTERPRETERS_WAMR_AOT=y\\nCONFIG_INTERPRETERS_WAMR_FAST=y\\nCONFIG_INTERPRETERS_WAMR_LIBC_BUILTIN=y\\n",
           "CONFIG_INTERPRETERS_WAMR=y\\nCONFIG_INTERPRETERS_WAMR_AOT=y\\nCONFIG_INTERPRETERS_WAMR_FAST=y\\n",
+          "CONFIG_INTERPRETERS_WAMR=y\\nCONFIG_INTERPRETERS_WAMR_AOT=y\\nCONFIG_INTERPRETERS_WAMR_CLASSIC=y\\n",
           "CONFIG_INTERPRETERS_WAMR=y\\nCONFIG_INTERPRETERS_WAMR_AOT=y\\nCONFIG_INTERPRETERS_WAMR_LIBC_BUILTIN=y\\n",
           "CONFIG_INTERPRETERS_WAMR=y\\nCONFIG_INTERPRETERS_WAMR_AOT=y\\n",
+          "CONFIG_INTERPRETERS_WAMR=y\\nCONFIG_INTERPRETERS_WAMR_FAST=y\\n",
+          "CONFIG_INTERPRETERS_WAMR=y\\nCONFIG_INTERPRETERS_WAMR_CLASSIC=y\\n",
         ]
 
     steps:
       - name: Install Toolchains
-        run: sudo apt install -y gcc-arm-none-eabi kconfig-frontends-nox genromfs
+        run: sudo apt install -y gcc-arm-none-eabi gcc-riscv64-unknown-elf kconfig-frontends-nox genromfs
```

Comment:
The toolchain `gcc-arm-none-eabi` and `gcc-riscv64-unknown-elf` are installed in every build, it may take some time but they are not required in each build: for x86 target, they are not needed, for arm target, gcc-riscv64 isn't needed, for riscv target, gcc-arm isn't needed. Shall we add conditions to control that? For example, split this into three steps:
step 1: install other toolchains for every target,
step 2: install gcc-arm-none-eabi, add a condition line:
```
if ${{ matrix.nuttx_board_config == "raspberrypi-pico:nsh" || matrix.nuttx_board_config == ... }}
``` 
step 3: install gcc-riscv64-unknown-elf, add a similar condition line
//////////////
Or install toolchain with a step, and set the install packages by condition sentences, not sure whether it works.

Reply:
OK, I'll try to figure it out.